src/main/java/net/datafaker/providers/base/Locality.java

```diff
@@ -53,52 +53,49 @@ public List<String> allSupportedLocales(Set<String> fileMasks) {
         Set<String> langs = Set.of(Locale.getISOLanguages());
         String[] paths = ManagementFactory.getRuntimeMXBean().getClassPath().split(File.pathSeparator);
         Set<String> locales = new HashSet<>();
-        for (String s: paths) {
-            try {
-                final SimpleFileVisitor<Path> visitor = new SimpleFileVisitor<>() {
-
-                    @Override
-                    public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
-                        final String filename;
-                        if (Files.isDirectory(file) || Files.isHidden(file) || !Files.isReadable(file)) {
-                            return super.visitFile(file, attrs);
-                        }
-                        if ((filename = file.getFileName().toString().toLowerCase(Locale.ROOT)).endsWith(".yml") || filename.endsWith(".yaml")) {
-                            final String parentFileName = file.getParent().getFileName().toString();
-                            if (langs.contains(parentFileName)) {
-                                locales.add(parentFileName);
-                            } else {
-                                locales.add(filename.substring(0, filename.indexOf('.')));
+        final SimpleFileVisitor<Path> visitor = new SimpleFileVisitor<>() {
+
+            @Override
+            public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
+                final String filename;
+                if (((filename = file.getFileName().toString().toLowerCase(Locale.ROOT)).endsWith(".yml") || filename.endsWith(".yaml")) && Files.isRegularFile(file) && Files.isReadable(file)) {
```

Comment:
Move the assignment from the condition to the declaration? (I think that might just simplify how it reads for future maintenance).

Reply:
makes sense